gui/src/components/logs/Logs.svelte

```diff
@@ -1,30 +1,63 @@
 <script lang="ts">
+  import { afterUpdate, beforeUpdate } from 'svelte';
+
   import FormattedLogMessage from './FormattedLogMessage.svelte';
-  import type { Log } from './logTypes';
+  import type { LogEvent } from './types';
+
+  export let events: LogEvent[] = [];
+
+  // Automatically scroll on new logs if the user is already scrolled close to the bottom of the content.
+  let logViewport: HTMLElement;
+  let wasScrolledCloseToBottom = true;
+  // Record whether the user was scrolled close to the bottom before new entries arrived.
+  // If so, scroll them to the new bottom after the update.
+  beforeUpdate(() => {
+    if (!logViewport) {
+      return;
+    }
 
-  export let logs: Log[];
+    const threshold = 150;
+    const currentPosition = logViewport.scrollTop + logViewport.offsetHeight;
+    const height = logViewport.scrollHeight;
+    wasScrolledCloseToBottom = currentPosition > height - threshold;
+  });
+
+  afterUpdate(async () => {
+    if (wasScrolledCloseToBottom && logViewport) {
+      logViewport.scrollTop = logViewport.scrollHeight;
+    }
+  });
 </script>
 
-<table>
-  {#each logs as log (log.id)}
-    <tr style={log.style}>
-      <td
-        class="time"
-        on:click={() => {
-          window.alert(`Full timestamp: ${log.time.full}`);
```

Comment:
I missed this in @jwmza's recent PR, but it looks like 9009f192b483adc0b08a589a3ee5bf0c38ab001c undid some of the work that I did in 9009f192b483adc0b08a589a3ee5bf0c38ab001c - Why?

Reply:
The floating timestamp was causing some usability issues with copy/paste as well as text formatting & whitespace issues. Unfortunately an in-situ solution is not really viable due to the visual overflow of the log table component, so it needs some way to render ontop of the application in some sort of messages/alerts layer.

The other solution is a toggle which toggles all the timestamps to full timestamps, which wouldn't cause the issues listed above.

Relocated full timestamp to `window.alert` in the meantime.